File: profiling/utils/tmux.py
import subprocess


class Tmux:
    def __init__(self) -> None:
        pass

    def create_session(session_name: str) -> None:
        subprocess.run(f'tmux new-session -d -s {session_name}', shell=True)
    
    def kill_session(session_name: str) -> None:
        subprocess.run(f'tmux kill-session -t {session_name}', shell=True)
    
    def run_command_in_session(session_name: str, command: str) -> None:
        subprocess.run(f'tmux send-keys -t {session_name} "{command}" Enter', shell=True)

    def capture_last_stdout_line(session_name: str) -> str:
        command = f'tmux capture-pane -t {session_name} -p'
        result = subprocess.run(command, shell=True, capture_output=True, text=True)
        lines = result.stdout.strip().split('\n')
        if lines:
            number = lines[-1]
        else:
            raise ValueError("No output captured")
        
        return number

File: utils/tmux.go
package utils

import (
	"fmt"
    "os/exec"
	"strings"
)

type Tmux struct{}

func (t *Tmux) CreateSession(sessionName string) {
    exec.Command("tmux", "new-session", "-d", "-s", sessionName).Run()
}

func (t *Tmux) KillSession(sessionName string) {
    exec.Command("tmux", "kill-session", "-t", sessionName).Run()
}

func (t *Tmux) RunCommandInSession(sessionName, command string) {
    exec.Command("tmux", "send", "-t", sessionName, fmt.Sprintf(`"%s"`, command), "Enter").Run()
}

func (t *Tmux) CaptureLastStdoutLine(sessionName string) (string, error) {
    cmd := exec.Command("tmux", "capture-pane", "-t", sessionName, "-p")
    output, err := cmd.Output()
    if err != nil {
        return "", err
    }
    lines := strings.Split(strings.TrimSpace(string(output)), "\n")
    return lines[len(lines)-1], nil
}
